python/dcr_template.py

```diff
@@ -287,67 +298,71 @@ def __init__(self, repository=".", obsid_range=None, band_name='g', wavelength_s
         self.use_psf = use_psf
         self.use_fft = use_fft
         self.debug = debug_mode
-        self._build_regularization()
+        self.regularize = self._build_regularization(x_size=self.kernel_size,
+                                                     y_size=self.kernel_size, n_step=self.n_step)
         self._calc_psf_model()
 
-    def _build_regularization(self):
+    @staticmethod
+    def _build_regularization(x_size=None, y_size=None, n_step=None):
         """
         Regularization adapted from Nate Lust's DCR Demo iPython notebook.
         Calculate a difference matrix for regularization as if each wavelength were a pixel, then scale
         the difference matrix to the size of the number of pixels times number of wavelengths
         """
-        # baseReg = _difference(self.kernel_size)
-        # reg_pix = np.zeros((self.n_step * self.kernel_size, self.n_step * self.kernel_size))
+        # baseReg = _difference(kernel_size)
+        # reg_pix = np.zeros((n_step * kernel_size, n_step * kernel_size))
 
-        # for i in range(self.n_step):
-        #     reg_pix[i::self.n_step, i::self.n_step] = baseReg
-        x_size = self.kernel_size
-        y_size = self.kernel_size
+        # for i in range(n_step):
+        #     reg_pix[i::n_step, i::n_step] = baseReg
         reg_pix = None
+        reg_lambda = None
+        reg_lambda2 = None
 
-        # reg_pix_x = np.zeros((self.n_step * x_size * y_size,
-        #                       self.n_step * x_size * y_size - x_size))
-        # for _ij in range(self.n_step * x_size * y_size - x_size):
+        # reg_pix_x = np.zeros((n_step * x_size * y_size,
+        #                       n_step * x_size * y_size - x_size))
+        # for _ij in range(n_step * x_size * y_size - x_size):
         #     reg_pix_x[_ij, _ij] = 1
         #     reg_pix_x[_ij + x_size, _ij] = -1
         # reg_pix_x = np.append(reg_pix_x, -reg_pix_x, axis=1)
 
-        # reg_pix_y = np.zeros((self.n_step * x_size * y_size,
-        #                       self.n_step * x_size * y_size - 1))
-        # for _ij in range(self.n_step * x_size * y_size - 1):
+        # reg_pix_y = np.zeros((n_step * x_size * y_size,
+        #                       n_step * x_size * y_size - 1))
+        # for _ij in range(n_step * x_size * y_size - 1):
         #     reg_pix_y[_ij, _ij] = 1
         #     reg_pix_y[_ij + 1, _ij] = -1
         # reg_pix_y = np.append(reg_pix_y, -reg_pix_y, axis=1)
         # reg_pix = np.append(reg_pix_x, reg_pix_y, axis=1)
 
         # # Extra regularization that we force the SED to be smooth
-        reg_lambda = np.zeros((self.n_step * x_size * y_size, (self.n_step - 1) * self.kernel_size**2))
-        for _f in range(self.n_step - 1):
+        reg_lambda = np.zeros((n_step * x_size * y_size, (n_step - 1)*x_size*y_size))
```

Comment:
(nitpick...) it's cleaner to have spaces around your multiplication operators, same as the rest of the line: 
```) * x_size * y_size)```